flytekit/core/node.py

```diff
@@ -21,6 +21,8 @@ def __init__(
         upstream_nodes: List[Node],
         flyte_entity: Any,
     ):
+        if id is None:
```

Comment:
@wild-endeavor this line seems odd?

Reply:
what do you mean?  like why would it ever be None?